src/test/kotlin/cacophony/semantic/analysis/VariablesMapCreationTest.kt

```diff
@@ -0,0 +1,129 @@
+package cacophony.semantic.analysis
+
+import cacophony.block
+import cacophony.controlflow.Variable
+import cacophony.intArg
+import cacophony.intFunctionDefinition
+import cacophony.intType
+import cacophony.lit
+import cacophony.lvalueFieldRef
+import cacophony.rvalueFieldRef
+import cacophony.struct
+import cacophony.structType
+import cacophony.testPipeline
+import cacophony.typedArg
+import cacophony.variableDeclaration
+import cacophony.variableUse
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.util.Map.entry
+
+class VariablesMapCreationTest {
+    @Test
+    fun `variable is created for let definitions`() {
+        // given
+        val xDef = variableDeclaration("x", lit(1))
+        val xUse = variableUse("x")
+        val ast = block(xDef, xUse)
+
+        // when
+        val variables = testPipeline().createVariables(ast)
```

Comment:
I don't know if usage of `testPipeline()` here is the best, as it calls resolveNames and checkTypes underneath - it shouldn't be much cost to write their result by hand, and we would have isolated test